### PR TITLE
Add unloadCSS() to the plugin api

### PR DESCRIPTION
**Pull request by nycex**
Title says everything.

### CODE REVIEW

File: components/plugin.js [hunk 1]
```diff
@@ -354,6 +354,39 @@ class Plugin extends EventEmitter {
       this.error('Failed to import css file', this._name, err)
     }
   }
+
+  async unloadCSS (file, watch = false) {
+    if (watch && !this.watcher) {
```
Comment (cking) [inline components/plugin.js]: does not need to initialize watcher on unload

File: components/plugin.js [hunk 2]
```diff
@@ -354,6 +354,39 @@ class Plugin extends EventEmitter {
...
+        )}"]`
+      )
+      if (el) {
+        this.info('Unloading css file', file)
+        while (el.firstChild) {
+          el.firstChild.remove()
+        }
+      }
+
+      if (watch) {
```
Comment (cking) [inline components/plugin.js]: use `watch && this.watcher` instead

File: components/plugin.js [hunk 3]
```diff
@@ -354,6 +354,39 @@ class Plugin extends EventEmitter {
...
+    try {
+      let el = document.querySelector(
+        `style[data-plugin="${this._id}"][data-filename="${file.replace(
+          /\\/g,
+          '\\\\'
+        )}"]`
+      )
+      if (el) {
+        this.info('Unloading css file', file)
+        while (el.firstChild) {
```
Comment (cking) [inline components/plugin.js]: as you unload anyway, just use `el.remove()`

File: components/plugin.js [hunk 4]
```diff
@@ -354,6 +354,39 @@ class Plugin extends EventEmitter {
       this.error('Failed to import css file', this._name, err)
     }
   }
+
+  async unloadCSS (file, watch = false) {
```
Comment (cking) [inline components/plugin.js]: make file optional, so that it unloads everything. then replace the call in line 57 with this.

**Comment by nycex** [comment]
Done.

File: components/plugin.js [hunk 5]
```diff
@@ -356,31 +354,34 @@ class Plugin extends EventEmitter {
   }
 
   async unloadCSS (file, watch = false) {
```
Comment (cking) [inline components/plugin.js]: now that i think about it, the watch can be removed.
just check for the watcher, no need to check for the watch param.
also please provide the optional file parameter (`file = null` or `file = false`)

**Comment by nycex** [comment]
Now really done :D

**Pull request merged**